sockpuppet/reflex.py

```diff
@@ -1,30 +1,85 @@
+from collections import UserDict
 from django.urls import resolve
 from urllib.parse import urlparse
 
 from django.test import RequestFactory
 
 PROTECTED_VARIABLES = [
     "consumer",
+    "context",
     "element",
+    "params",
     "selectors",
     "session",
     "url",
 ]
 
 
+class Context(UserDict):
+    """
+    A dictionary that keeps track of whether it's been used as dictionary
+    or if values has been set with dot notation. We expect things to be set
+    in dot notation so a warning is issued until next major version (1.0)
+    """
```

Comment:
As a maintainer this is a nice message to help explain what the purpose of this data type is, but as a user I would be surprised to get this message when doing `help(reflex.context)`.

Reply:
That is a very good point.